### ISSUE TITLE
Ignoring settings

### ISSUE THREAD

**Issue by EpsilonKu** [SidOfc/carbon.nvim#113]
Seems like it has problem with [`lazy.nvim`](https://github.com/folke/lazy.nvim). It ignore settings from `setup`.
I run minimal init with `nvim --clean -u init.lua`:
```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

require("lazy").setup({
  "folke/which-key.nvim",
  {"SidOfc/carbon.nvim", config = function ()
    require('carbon').setup({
          sync_pwd = true,
	  auto_open = false,
          indicators = { collapse = '▾', expand = '▸' },
        })
  end}
})
```
It automaticall opens and indicators doesn't changed. 
`nvim` version:
```lua
NVIM v0.9.0-dev-5e1687e-dirty
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions     -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security     -fstack-clash-protection -fcf-protection -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/epsku/.cache/paru/clone/neovim-git/src/build/cmake.config -I/home/epsku/.cache/paru/clone/neovim-git/src/neovim-git/src -I/usr/include -I/home/epsku/.cache/paru/clone/neovim-git/src/build/src/nvim/auto -I/home/epsku/.cache/paru/clone/neovim-git/src/build/include
Compiled by epsku

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```
OS: `Arco linux`, updated to latest rolling-release.

**Comment by SidOfc** [comment]
I'm thinking this is a package manager problem, did you try opening an issue there already?
Otherwise I'll also need some information as to what this achieves since I'm not familiar with this plugin manager.

**Comment by EpsilonKu** [comment]
Yeah, seems like `lazy.nvim` issue. Opened https://github.com/folke/lazy.nvim/issues/271

**Comment by SidOfc** [comment]
Good stuff @EpsilonKu, cheers for sharing! I will close this for now until an issue on Carbon's side is identified.

Happy new year 👍

**Comment by EpsilonKu** [comment]
Damn, u fast. :skull:

**Comment by jemag** [comment]
@SidOfc `Lazy.nvim` loads `init.lua` then `plugin/` then config.

I believe your plugin will ignore the settings because once setup is reached, `data.initialized` is already equal to `true`, since `initialize` is already called from `plugin/`

Folke seems to mention that this is the proper order of how it should go here : https://github.com/folke/lazy.nvim/issues/123#issuecomment-1367773015

**Comment by SidOfc** [comment]
@jemag ah I see now, indeed there is some ambiguity in the way initialization is handled. Reason being that `carbon.initialize` assumes that `carbon.setup` is called _before_ `carbon.initialize`. This is not the case however, when the `plugin/` directory is loaded before the `carbon.setup` method is executed.

This behavior mainly exists to prevent calling `setup` twice but I think I can tweak it fairly easily to allow some different load orders.

Considering the issue has been identified I'm going to reopen this issue, additionally, I think https://github.com/folke/lazy.nvim/issues/271 can be closed. Thank you for chiming in with this!

**Comment by SidOfc** [comment]
@EpsilonKu I just released a patch which introduces a global `g:carbon_lazy_init` variable. When set to `true` or `1`, the call to `carbon.initialize` will be postponed until after `carbon.setup` has been called.

I've checked that this worked without any plugin manager though, so I'll leave this open for you to confirm the fix also works with lazy.nvim :)

At any rate, cheers for the report, and once again @jemag thanks for the additional information!
